safe_transaction_service/tokens/migrations/0004_ethereum_address_field_v2_20211201_1512.py

```diff
@@ -13,7 +13,10 @@ class Migration(migrations.Migration):
 
     operations = [
         migrations.RunSQL(
-            "DROP INDEX IF EXISTS tokens_token_address_18ef94ca_like",
+            """
+            DROP INDEX IF EXISTS tokens_token_address_18ef94ca_like;
+            ALTER TABLE "tokens_token" ALTER COLUMN "address" TYPE bytea USING DECODE(SUBSTRING("address", 3), 'hex');
```

Comment:
:thinking: what is this doing?

Reply:
why substring 3?

Reply:
Currently we are storing addresses as `varchar` with `0x` prefix. `0x` prefix is not valid for the postgres `DECODE` function. So this line is converting current `varchar` addresses to `bytea` binary ones

Reply:
Apparently SQL uses 1 based indexes .... this is what confused me: https://www.w3schools.com/sql/trymysql.asp?filename=trysql_func_mysql_substring

![image](https://user-images.githubusercontent.com/2896048/146190608-6d4636a9-3291-42b8-af4a-d0d1ee5e4558.png)